src/main/java/com/jcabi/ssh/SSH.java

```diff
@@ -182,6 +193,17 @@ public SSH(final String adr, final int prt,
     }
 
     /**
+     * Constructor using SSHOptions object.
+     * @param options SSHOptions options object with connection properties
+     * @throws UnknownHostException If fails
+     */
+    public SSH(final SSHOptions options) throws UnknownHostException {
```

Comment:
@LinkedList let's keep it simple. Please, remove `SSHOptions` and add one more constructor, which accepts `passphrase` as well as other params. You can suppress `ParameterNumberCheck` with
`@checkstyle ParameterNumberCheck (n lines)` in constructor's javadoc